data/repository/src/commonMain/kotlin/social/androiddev/common/repository/timeline/HomeTimelineRepository.kt

```diff
@@ -0,0 +1,37 @@
+package social.androiddev.common.repository.timeline
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import org.mobilenativefoundation.store.store5.Market
+import org.mobilenativefoundation.store.store5.MarketResponse
+import org.mobilenativefoundation.store.store5.ReadRequest
+import social.androiddev.common.timeline.TimelineItem
+
+interface HomeTimelineRepository {
+    suspend fun read(): Flow<MarketResponse<List<TimelineItem>>>
+}
```

Comment:
Unless we want to change this, the interface should be defined in `domain` layer/module. The ui/feature modules only depend on `domain` and not on any repository module directly. repository module(s) depend on domain module(s) and can then extend the relative interface

Reply:
Sorry not following, If the real impl is in the data/repository layer how can the interface be in the domain layer?

Reply:
so `data/repo*` _depends on_ `domain`
`domain` contains: UseCases/simple repo wrappers, domain models, and interfaces for repositories
`data/repo*` provides the impls for the interfaces in `domain`
`data/repo*` has a koin module which binds the impl to the interface

ui/feature _depends on_ `domain` (not data/repo) (i.e. the ViewModels, Components, etc have the UseCases/Repo wrappers injected and do not have access to any data/repo)